otx/algorithms/classification/utils/data.py

```diff
@@ -115,6 +120,15 @@ def __init__(
                 dataset_item = DatasetItemEntity(image, annotation_scene, subset=subset)
                 dataset_items.append(dataset_item)
 
+        if ul_data_root is not None:
+            if not self.data_type == ClassificationType.MULTIHEAD:
```

Comment:
Do we have specific issue for multihead type?

Reply:
As far as I know, we don't support semi for hierarchical, so disabled to get unlabeled cli arg.

Reply:
Then, I prefer to leave a log message that can notify the the user. i.e.) "In hierarchical classification, semi-SL training is automatically converted to supervised training"

Reply:
Oh that error log in _init_recipe when it tries to load semi recipe but it's hierarchical. It would be better we log that warning msg in that code.